src/TriangleNodesProvisioner.cpp

```diff
@@ -441,8 +444,116 @@ namespace blitzdg {
     }
     
     void TriangleNodesProvisioner::buildMaps() {
-        //const MeshManager& meshMgr = *Mesh2D;
-        // WIP here...
+        firstIndex ii;
+        secondIndex jj;
+        thirdIndex kk;
+
+        index_matrix_type nodeIds(NumLocalPoints, NumElements);
+        
+        const index_matrix_type& Fmsk = *Fmask;
+        index_vector_type& vmM = *vmapM;
+        index_vector_type& vmP = *vmapP;
+        index_vector_type& mP = *mapP;
+
+        const index_vector_type& E2E = Mesh2D.get_EToE();
+        const index_vector_type& E2F = Mesh2D.get_EToF();
+        const index_vector_type& E2V = Mesh2D.get_Elements(); 
+
+        real_matrix_type xmat(NumLocalPoints, NumElements, ColumnMajorOrder());
+        real_matrix_type ymat(NumLocalPoints, NumElements, ColumnMajorOrder());
+
+        xmat = *xGrid; ymat = *yGrid;
+
+        real_vector_type x(NumElements*NumLocalPoints);
+        real_vector_type y(NumElements*NumLocalPoints);
+        reshapeMatTo1D(xmat, x.data(), false);
+        reshapeMatTo1D(ymat, y.data(), false);
+
+        // Assemble global volume node numbering.
+        nodeIds = ii + NumLocalPoints*jj;
+
+        Array<index_type, 3> vmapM3(NumFacePoints, NumFaces, NumElements);
+        Array<index_type, 3> vmapP3(NumFacePoints, NumFaces, NumElements);
+        Array<index_type, 3> mapP3(NumFacePoints, NumFaces, NumElements);
+
+        vmapM3 = 0*kk; vmapP3 = 0*kk; mapP3 = 0*kk;
+
+        for (index_type k=0; k < NumElements; ++k) {
+            for (index_type f=0; f < NumFaces; ++f) {
+                for (index_type n=0; n < NumFacePoints; ++n) {
+                    vmapM3(n,f,k) = nodeIds(Fmsk(n,f), k);
+                }
+            }
+        }
```

Comment:
Since blitz stores arrays in row-major order by default you should change the nesting order of these loops to achieve better performance. Assuming that `nodeIds` and `Fmsk` are also row-oriented, then the optimal nesting would be `n`, then `f`, and then `k` as the innermost loop.

Reply:
Okay, I've swapped the order in one place, but could not do it in a few other places where it affects the write-order of the 1d arrays.

Right now, the goal is to match the ordering of these map arrays in the NUDG matlab code until we have a working solver. Once we do, I will attempt to come back and refactor so everything is in row-major ordering.

We can also fix the problem in the 1D solvers where we have 'uCol', and 'u' around (column and row-major copies of the same variable), by switching everything over to row-major as you've suggested to me in the past.